components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+
+        try {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Authentication Request has initialized");
+            }
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            if (authenticatorProperties != null) {
+
+                String envUAEPass = getUAEPassEnvironment(context);
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("UAEPass " + envUAEPass + " Environment Selected");
+                }
+                String clientId = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.CLIENT_ID);
+                String authorizationEP = getAuthorizeUrl(envUAEPass);
+                String callBackUrl = authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.CALLBACK_URL);
+                String state = context.getContextIdentifier() + "," + UAEPassAuthenticatorConstants.UAE.LOGIN_TYPE;
+
+                OAuthClientRequest authzRequest = OAuthClientRequest
+                        .authorizationLocation(authorizationEP)
+                        .setClientId(clientId)
+                        .setRedirectURI(callBackUrl)
+                        .setResponseType(UAEPassAuthenticatorConstants.UAE.OAUTH2_GRANT_TYPE_CODE)
+                        .setState(state)
+                        .buildQueryMessage();
+
+                if(LOG.isDebugEnabled()){
+                    LOG.debug("Authorization request build with must required query parameters.");
+                }
+                String loginPage = authzRequest.getLocationUri();
+
+                if (StringUtils.isNotBlank(authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.QUERY_PARAMS))) {
+                    loginPage = processAdditionalQueryParamSeperation(authenticatorProperties, loginPage);
+                } else {
+                    Map<String, String> paramMap = new HashMap<>();
+                    paramMap.put(UAEPassAuthenticatorConstants.UAE.ACR_VALUES,
+                            UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.DEFAULT_ACR_VALUES);
+                    paramMap.put(UAEPassAuthenticatorConstants.UAE.SCOPE,
+                            UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.DEFAULT_SCOPES);
+                    loginPage = FrameworkUtils.buildURLWithQueryParams(loginPage, paramMap);
+                }
+
+                response.sendRedirect(loginPage);
+            } else {
+                throw new AuthenticationFailedException("Error while retrieving properties. "
+                        + "Authenticator Properties cannot be null");
+            }
+
+        } catch (IOException e) {
+            LOG.error("Authorization request building failed.");
+            throw new AuthenticationFailedException("Unable to pick correct env or problem occurred in additional " +
+                    "query params when building the authorize request", e);
```

Comment:
Above AuthenticationFailedException is chained until to the carbon identity framework. Once this exception throws, it will pass to the AbstractApplicationAuthenticator, and then it will navigate to the DefaultStepHandler class of the carbon identity framework. BTW the stack trace will be printed as a DEBUG log.[1] The reason for adding such a log error message is, to identify the exact point where the exception is getting thrown. (If there is not having proper exception chain in the flow of the framework, the exact cause will not be getting traced in the carbon log. For the best cases, I have added the error log as well. Then at the beginning of the stack trace the error log message will be printed.)

[1] - https://github.com/wso2/carbon-identity-framework/blob/master/components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java#L732-L745